GameBoard/Scripts/Autocomplete/autocompleteFriends.ts

```diff
@@ -0,0 +1,126 @@
+
+                if (targetElement && !this.resultSource.contains(targetElement)) {
+                    this.closeAllAutocompleteResults();
+                }
+            });
+    }
+
+    private newResultsTimeout() {
+        if (this.currentTimeout) {
+            clearTimeout(this.currentTimeout);
+        }
+        this.currentTimeout = setTimeout(() => this.showAutocompleteResults(), this.timeoutDuration);
+    }
+
+    showAutocompleteResults() {
+        const value = this.source.value;
+
+        this.closeAllAutocompleteResults();
+
+        if (!value || value.length < this.minimalCharactersThreshold) {
+            return;
+        }
+
+        fetch(`${this.getUrl}?input=${value}&userName=${this.userName}&groupId=${this.groupId}`,
+                {
+                    method: "GET",
+                })
+            .then(response => this.createAutocompleteResults(response))
+            .catch((reason) => this.createErrorResult(reason));
+    }
+
+    private createAutocompleteResults(response: Response) {
+        response.text()
+            .then(responseText => {
+                if (!response.ok) {
+                    this.createErrorResult(responseText);
+                    return;
+                }
+
+                const template = document.createElement("template");
+                template.innerHTML = responseText.trim();
+                const resultDiv = template.content.firstChild as HTMLElement;
+
+                if (!resultDiv) {
+                    console.error("Autocomplete - could not create a results div from response.");
+                }
+
+                this.resultSource.prepend(resultDiv);
+
+                $(resultDiv)
+                    .mCustomScrollbar({
+                        theme: "minimal"
+                    });
+
+                this.setAutocompleteResultsPosition(resultDiv);
+            })
+            .catch(reason => this.createErrorResult(reason));
+    }
+
+    private createErrorResult(reason: any) {
+        console.error(reason);
+
+        const errorDiv = document.createElement("div");
+        errorDiv.classList.add("user-search-error", "text-danger", `${this.autocompleteResultsClass}`);
+
+        const text = document.createElement("p");
+        text.innerText = "There is an issue with the search service. Please, try again later.";
+
+        errorDiv.appendChild(text);
+        this.resultSource.prepend(errorDiv);
+
+        this.setAutocompleteResultsPosition(errorDiv);
+    }
+
+    private setAutocompleteResultsPosition(result: HTMLElement) {
+        const height = result.offsetHeight;
+        const defaultTop = window.getComputedStyle(result).top;
+        const defaultTopAsInt = defaultTop ? parseInt(defaultTop) : 0;
+
+        result.style.top = `${defaultTopAsInt - height}px`;
+    }
+
+    closeAllAutocompleteResults() {
+        document.querySelectorAll(`.${this.autocompleteResultsClass}`).forEach((el) => el.remove());
+    };
+}
```

Comment:
What the actual fuck?

You copy-pasted the entire autocomplete and changed ONE SINGLE LINE with the fetch to a different url. How you could even think anyone would accept this as a solution is beyond me.

Reply:
Yeah, I agree, it's fixed now